hugr/src/extension/op_def.rs

```diff
@@ -635,4 +635,34 @@ mod test {
         );
         Ok(())
     }
+
+    #[test]
+    fn instantiate_extension_delta() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::extension::prelude::{BOOL_T, PRELUDE_REGISTRY};
+
+        let mut e = Extension::new(EXT_ID);
+        let ext_id = e.name().clone();
+
+        let es = ExtensionSet::singleton(&ext_id);
+        let params: Vec<TypeParam> = vec![TypeParam::Extensions];
+        let args = [TypeArg::Extensions { es: es.clone() }];
+        let mut db_set = ExtensionSet::new();
+        db_set.insert_type_var(0);
+        let fun_ty = FunctionType::new_endo(vec![BOOL_T]).with_extension_delta(db_set);
+
+        let exp_fun_ty = FunctionType::new_endo(vec![BOOL_T]).with_extension_delta(es);
```

Comment:
optional/super-nit: I think I'd finish making `def` before you start on what you're going to do with `def` (so move `let es` and `let exp_fun_ty` after `let def`)